frontend/src/services/auth.js

```diff
@@ -27,16 +27,27 @@ export const getToken = () => {
   return localStorage.getItem('jwt')
 }
 
-export const isAuthenticated = () => {
+export const isAuthenticated = async () => {
   const token = getToken()
 
   if (!token) return false
 
   try {
     const decoded = JSON.parse(atob(token.split('.')[1]))
     const exp = decoded.exp
+    const hasExp = exp > Date.now() / 1000;
 
-    return exp > Date.now() / 1000
+    if (!hasExp) return true
+
+    const res = await Api.post('/auth/token').send({ refresh: localStorage.getItem('refresh') })
+
+    if (res.status == 401) return false
+
+    const data = await res.json()
+
+    localStorage.setItem('jwt', data.token)
+
+    return true
```

Comment:
Create a function to check if the session update is possible, and if so, update it